sdk/python/kfp/dsl/_metadata.py

```diff
@@ -48,10 +48,15 @@ def to_dict_or_str(self):
   @staticmethod
   def from_dict_or_str(json):
     type_meta = TypeMeta()
+    if isinstance(json, str) and '{' in json:
+      import ast
+      json = ast.literal_eval(json)
     if isinstance(json, dict):
       if not _check_valid_type_dict(json):
         raise ValueError(json + ' is not a valid type string')
       type_meta.name, type_meta.properties = list(json.items())[0]
+      # Convert possible OrderedDict to dict
+      type_meta.properties = dict(type_meta.properties)
```

Comment:
This code will always override the properties that has been set on the line above. Is it correct?

Reply:
The reason to convert the OrderedDict to dict is that component module generates the ordereddict and the serialization of the orderedDict is hard to de-serialize when the pipelineparam string is matched to our regex pattern.